### PR TITLE
Enable complex python tests

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Complex is still disabled by default. This PR temporarily add an env `PYTORCH_NVFUSER_ENABLE_COMPLEX_PY` to enable complex dtype in python

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
I am seeing a lot of `test_nvfuser_correctness` failures, some are complex, some are not. Other than those, tests should be fine.

File: test/test_jit_cuda_fuser.py
Comment (jjsjann123) [inline test/test_jit_cuda_fuser.py]: Are we not checking outputs for bfloat16?

File: torch/csrc/jit/codegen/cuda/parser.cpp
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/parser.cpp]: nitpick, we can do nullptr to avoid nolint.

File: torch/csrc/jit/codegen/cuda/partition.cpp
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/partition.cpp]: I'm gonna suggest that we follow the pattern here: #1666.

maybe we can rebase this on Naoya's PR or wait for that one to be merged first?

File: torch/csrc/jit/codegen/cuda/partition.cpp
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/partition.cpp]: Nice~

**Review by jjsjann123 — DISMISSED** [review]
LGTM.

Some minor nitpicks. One thing I'd prefer is to have the env variable thing to be refactored on #1666

**Comment by jjsjann123** [comment]
I'm also pretty open to merge this as-is and refactor the env variable in a separate PR after #1666 merged. So feel free to yell at me to get a stamp :wink:

**Comment by zasdfgbnm** [comment]
This is not urgent, I can wait for https://github.com/csarofeen/pytorch/pull/1666.

File: torch/csrc/jit/codegen/cuda/partition.cpp
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/partition.cpp]: Looks like the https://github.com/csarofeen/pytorch/pull/1666 assumes enabled by default? Here I want it to be disabled by default. Do you think I should add a similar mechanism `PYTORCH_NVFUSER_ENABLE`, or should I keep this flag as is?

File: test/test_jit_cuda_fuser.py
Comment (zasdfgbnm) [reply test/test_jit_cuda_fuser.py]: Yeah, bfloat16 is giving the following error on my system:
```
failing case torch.bfloat16
<built-in method exp of type object at 0x7fa559f971c0>
tensor([    -inf, -10.0000,  -3.1406,  -1.0000,  -0.5000,   0.0000,   1.0000,
          0.5000,   3.1406,  10.0000,      inf,      nan], device='cuda:0',
       dtype=torch.bfloat16)
tensor([0.4004, 0.2559, 0.0859, 0.1943, 0.6953, 0.6250, 0.8477, 0.2070, 0.7227,
        0.8320, 0.3809, 0.9922], device='cuda:0', dtype=torch.bfloat16) False
dif    :  tensor([[ -0.1250],
        [-48.0000]], device='cuda:0', dtype=torch.bfloat16)
inp1   :  tensor([[   9.6250],
        [4080.0000]], device='cuda:0', dtype=torch.bfloat16)
inp2   :  tensor([[   9.7500],
        [4128.0000]], device='cuda:0', dtype=torch.bfloat16)
maximum difference tensor(-0.1250, device='cuda:0', dtype=torch.bfloat16)
```
That is strange, because I didn't touch bfloat16 at all.

Comment (zasdfgbnm) [reply test/test_jit_cuda_fuser.py]: Looks like this issue is fixed in https://github.com/csarofeen/pytorch/pull/1674

File: torch/csrc/jit/codegen/cuda/partition.cpp
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/partition.cpp]: Adding a `PYTORCH_NVFUSER_ENABLE` sounds good to me~ :pray:

**Comment by zasdfgbnm** [comment]
@jjsjann123 This PR should be ready

**Comment by zasdfgbnm** [comment]
ping @jjsjann123

File: torch/csrc/jit/codegen/cuda/parser.cpp
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/parser.cpp]: Agree with the nit `CgValue cg_val = nullptr;`

File: torch/csrc/jit/codegen/cuda/utils.cpp
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/utils.cpp:154]: Looks good, thanks.

**Review by csarofeen — APPROVED** [review]
LGTM only one minor nit left in here, but no big deal. Feel free to merge.

File: torch/csrc/jit/codegen/cuda/parser.cpp
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/parser.cpp]: Oops. I don't know why I missed this comment. I have updated my PR.

**Pull request merged**
